### DISCUSSION TITLE
BTCPay custom user role (10K USD Bounty)

### DISCUSSION THREAD

**Discussion by lucidddreams** [btcpayserver/btcpayserver#2650] [Ideas / Feature Requests]
Currently there are 2 user roles in BTCPay, guest and owner, In our use case where our user (with owner permission) that approves a pull payout, can just edit amounts, add new destination on a pull payment Send page and access the Send page with hot wallet  without restriction.

Feature request with 0.3 BTC ($10,000) bounty
Develop a custom access level to btcpayserver features on the following and its functions: 
1. Access to Send page  
2. Editing amounts & adding destination on "pull payout" Send page. To reproduce - > create pull payout, fill up the pull payout link, send the pull payout. (risk here is user can edit amounts and add destination)
3. Access to settings page - Risk level is high, user with owner permission can view seed
4. Other access levels necessary to improve security on shared store with hot wallet enabled.

Reward
0.15 BTC ($5,000) will be donated to the BTCPay foundation
0.15 BTC ($5,000) will be received by the contributor/maintainer with approved pull request.

I am willing to transparently transfer USD$10,000/0.3 BTC as escrow to a BTCPay representative for the reward.

**Comment by pavlenex** [comment]
Similar slightly to https://github.com/btcpayserver/btcpayserver/issues/442

**Comment by Kukks** [comment]
I will assist anyone who takes on this issue (I do not want to collect a share of this bounty). 

My initial thoughts on this are to make use of our Greenfield API permissions, expand them into more fine-grained controls and adjust the Ui to make use of them.

**Comment by bolatovumar** [comment]
This sounds like it can be expanded on further by allowing admins to create custom user roles similar to how you can create API keys with different permissions and then assign these roles to users. Then you can have one user who can only deal with pull payments, another user who can do something else entirely and so on and so forth.

I can't really commit to doing this but it doesn't sound like something that would be _hard_ to do. What @Kukks said about sounds like it should work.

**Comment by lucidddreams** [comment]
This bounty is still ON.

**Comment by ghost** [comment]
I'll take a crack at this.

**Comment by ghost** [comment]
We briefly discussed this on [today's call](https://github.com/btcpayserver/organization/issues/64). @NicolasDorier and others gave an impression that this will be quite a difficult undertaking, and probably a much higher complexity level than I had previously assumed.

It's possible that some were thinking of the fully-expanded version of this idea as @bolatovumar describes above, instead of the specific requirements described by @lucidddreams. 

@NicolasDorier, do you think a "minimum viable feature" PR to satisfy exactly (and only) the OP's requirements would be significantly simpler?

**Reply by lucidddreams** [reply]
![image](https://user-images.githubusercontent.com/8253590/139574404-a92b97d0-5615-4a3b-981e-8ac9c2c8a267.png)
@satwo yes, the description provided by @bolatovumar is the exact solution that we need.
